csrc/fusion.h

```diff
@@ -84,6 +84,31 @@ class FusionGuard {
   static thread_local Fusion* active_fusion_;
 };
 
+// Set the enum base to `int` so it can be safely serialized as a part of
+// serde::InputOutputAlias.
+enum class AliasType : int {
+  // For example, the tensor storing BatchNorm's running mean. The output, the
+  // updated EMA, is written to the same tensor as the input. Therefore, they
+  // are alias of the same shape but different data.
+  SameShapeDifferentData,
+  // For example, the input and the output of the same ViewOp are likely alias
+  // of different shapes but the same underlying data. In this case, we use
+  // `ExpressionEvaluator` (instead of a kernel) to compute the output tensor.
+  DifferentShapeSameData,
+};
```

Comment:
Is trivial forwarding supposed to be another AliasType, where we have same shape and same data which doesn't fit into the existing naming.

nitpick: the name feels a bit hard to read. Especially this one `SameShapeDifferentData`, where we are just describing an inplace-update.

Reply:
> where we are just describing an inplace-update.

I was looking for a better name! Yes, I'll change the names to `InplaceUpdate` and `PointerCast`.

Reply:
> Is trivial forwarding supposed to be another AliasType, where we have same shape and same data which doesn't fit into the existing naming.

Good point -- I'll treat it as a special type of `PointerCast`. `PointerCast` can be cheaply evaluated on host whereas the other type can't.

Reply:
Done.